### DISCUSSION TITLE
Pipeline CLI says the --data option is required, but it says "Job request is not valid" when I include it

### DISCUSSION THREAD

**Discussion by emassey0135** [daisy/pipeline#794] [Q&A]
I am trying to use the Pipeline CLI to convert some books with TTS, and now I can't even get it to start the job. When I run "help" for dtbook-to-daisy3, epub3-to-epub3, and dtbook-validator, they all say both the "--source" and "--data" options are required. When I run the scripts with only the "--source" and "-o" options, the CLI says the "--data" option is required, but when I pass a zip file to "--data" containing the entire DAISY book the dtbook was taken from, it says "Job request is not valid." I tried this on both Windows 11 and Arch Linux, and with both Java 8 and Java 21, and with the newest version (1.14.19), a few older versions (1.14.18 and 1.14.16), and the latest code from this repository, all with the same behavior. However, it was working a few months ago before I reinstalled the operating systems on my computer, so I'm not sure what I'm doing wrong. If it was a bug introduced after I used it last, then at least 1.14.16 should work.

**Comment by emassey0135** [comment]
After further investigation, I discovered that org.daisy.pipeline.ws.localfs is set to false by default, meaning that the Pipeline webservice does not access the local filesystem and needs a zip file of the files referenced by the dtbook. When I set it to true in pipeline.properties, the "--data" option is no longer required, and the job is created successfully. However, there still may be a bug, since passing a zip file to the "--data" option failed. Is there a certain directory structure the zip file must have, or limitations on the possible compression algorithms? Or should I report this as a bug?

**Reply by bertfrees** [reply]
@emassey0135 Could you please report an issue with the exact command and input file? And sorry for the late reply!
